### PR TITLE
NT-0014 - Activate user by email confirmation

### PR DESCRIPTION
**Pull request by mras2303**

### CODE REVIEW

File: routes/api/users.js [hunk 1]
```diff
@@ -14,4 +14,9 @@ router.post(
   usersController.registerUser
 );
 
+// @route   GET api/users/activate/:userId/:activationKey
+// @desc    Activate account
+// @access  Public
+router.get('/activate/:userId/:activationKey', usersController.activateUser);
```
Comment (damian0o) [inline routes/api/users.js]: I believe activationKey would be enough as a parameter.

File: controllers/users/users.js [hunk 2]
```diff
@@ -33,6 +70,7 @@ exports.registerUser = (req, res) => {
           if (err) throw err;
           // Store hash in your password DB.
           newUser.password = hash;
+          newUser.activationKey = Math.floor(Math.random()*10000000);
```
Comment (damian0o) [inline controllers/users/users.js]: We can use bcryptjs and generate some hash from user details and used it as activationKey.
If we would like to transport some additional information with activationKey itself and not to depend on database support we can use signed or encrypted JWT token.

File: controllers/users/users.js [hunk 3]
```diff
@@ -4,6 +4,43 @@ const jwt = require("jsonwebtoken");
...
 const User = require("../../models/User");
 const messages = require("../messages");
+const nodemailer = require("nodemailer");
+require("dotenv").config();
+
+const sendVerificationEmail = user => {
+  const transporter = nodemailer.createTransport({
+    service: "gmail",
+    auth: {
+      user: process.env.GMAIL_USER,
```
Comment (damian0o) [inline controllers/users/users.js]: How developers should deliver this environment ? 
Should we send this credentials to all active developers to make allow them test this functionality ?

Comment (mras2303) [reply controllers/users/users.js]: Will be good idea to store content of .env in slack section?

Comment (damian0o) [reply controllers/users/users.js]: For now not so bad ;)

File: controllers/users/users.js [hunk 4]
```diff
@@ -4,6 +4,43 @@ const jwt = require("jsonwebtoken");
...
+  }
+
+  transporter.sendMail(mailOptions, (err, data) => {
+    if(err) throw err;
+  })
+}
+
+exports.activateUser = async (req, res) => {
+  const { activationKey } = req.params;
+  await User.findOneAndUpdate({ activationKey }, { emailVerified: true })
```
Comment (damian0o) [inline controllers/users/users.js]: What should be returned if activation key is expired not exists or already used?

**Pull request merged**